adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -928,12 +928,28 @@ protected void setNonEmptyResponsePayload(final HttpServerResponse response, fin
                                         deviceId);
                                 commandContext.release();
                             } else {
-                                addMicrometerSample(commandContext, commandSample);
-                                // put command context to routing context and notify
-                                ctx.put(CommandContext.KEY_COMMAND_CONTEXT, commandContext);
-                                cancelCommandReceptionTimer(ctx);
-                                setTtdStatus(ctx, TtdStatus.COMMAND);
-                                responseReady.tryComplete();
+                                getTenantConfiguration(tenantId, commandContext.getTracingContext())
+                                        .compose(tenantObject -> checkMessageLimit(tenantObject,
+                                                command.getPayloadSize()))
+                                        .setHandler(result -> {
+                                            if (result.succeeded()) {
+                                                addMicrometerSample(commandContext, commandSample);
+                                                // put command context to routing context and notify
+                                                ctx.put(CommandContext.KEY_COMMAND_CONTEXT, commandContext);
+                                                cancelCommandReceptionTimer(ctx);
+                                                setTtdStatus(ctx, TtdStatus.COMMAND);
```

Comment:
IMHO we need to cancel the timer and set the status in both cases so these two lines should be moved down to where we complete the responseReady future

Reply:
I moved them.